atc/api/pipelineserver/archive_test.go

```diff
@@ -0,0 +1,91 @@
+package pipelineserver_test
+
+import (
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+
+	"github.com/concourse/concourse/atc/api/pipelineserver"
+	"github.com/concourse/concourse/atc/api/pipelineserver/pipelineserverfakes"
+	"github.com/concourse/concourse/atc/db/dbfakes"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+//go:generate counterfeiter code.cloudfoundry.org/lager.Logger
```

Comment:
this caught my attention, it's something we *seem* to only be doing here

```console
$  ~/workspace/concourse/atc $ ag -Q "lager.Logger" | grep counter
api/pipelineserver/archive_test.go:16://go:generate counterfeiter code.cloudfoundry.org/lager.Logger
```

do you think this is something we should adopt more? I'm curious about the rationale to testing the log statements here, and would challenge is a bit: if we were adding a `someMetric.Add(1)` (i.e., if we were incrementing a counter for something we want to measure), and even more, we had a trace that we wanted to capture, should we test those three things? log, metric, and trace?

(please don't take the above as criticism :sweat_smile: curious to know more about it)

thanks!

Reply:
I've spent a lot of time inspecting logs from Concourse with no access to the cluster. Based on this, I'm shocked by how often I really wish Concourse was logging something and I find that it's not. In general I haven't yet felt like it's necessary to unit-test every log message, but there are times and places where logging feels like a really desirable _feature_ with real _business value_ and I want a test to verify that it will definitely be happening - in such cases I'd be pretty upset if the log stopped happening haha.